src/fields.jl

```diff
@@ -109,16 +109,12 @@ FaceFieldZ(arch, g) = FaceFieldZ(zeros(arch, g), g)
 @inline lastindex(f::Field, dim) = lastindex(f.data, dim)
 @inline setindex!(f::Field, v, inds...) = setindex!(f.data, v, inds...)
 
+@inline data(f::Field) = f.data[1:f.grid.Nx, 1:f.grid.Ny, 1:f.grid.Nz]
```

Comment:
may be better to return `view(f.data, 1:Nx, 1:Ny, 1:Nz)`

Reply:
Yeah in hindsight that would be better. There are a few places where you don't want a view, but often you do want a view.

Reply:
RIght — but possibly we should add another function for copying data. Like, copydata.

Reply:
I believe we should expect a view to behave identically to the array pre-halo regions... ?

Reply:
Done.